craft_providers/multipass/_launch.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Multipass Provider."""
+
+import logging
+
+from craft_providers import Base
+
+from .multipass_instance import MultipassInstance
+
+logger = logging.getLogger(__name__)
+
+
+def launch(
+    name: str,
+    *,
+    base_configuration: Base,
+    cpus: int = 2,
+    disk_gb: int = 256,
+    image_name: str,
+    mem_gb: int = 2,
+) -> MultipassInstance:
+    """Create, start, and configure instance.
+
+    :param name: Name of instance.
+    :param base_configuration: Base configuration to apply to instance.
+    :param cpus: Number of CPUs.
+    :param disk_gb: Disk allocation in gigabytes.
+    :param image_name: Multipass image to use, e.g. snapcraft:core20.
```

Comment:
@saviq can we have a "craft" alias in a near future?

Reply:
So that `snapcraft:*` equals `craft:*`? Which one would we publicize going forward?

Reply:
We are going to use this for our other craft implementations (e.g.; charmcraft), so I guess calling it craft will have a longer run than other names.

We really don't expose the name unless there is an error, I suspect it will be confusing to say "Failed to launch snapcraft:core20" in a charmcraft error message. Speaking of which, if we don't already, creating aliases (I believe they exist on simplestreams) for corexx to Ubuntu releases (16.04, 18.04, ...).

Let's discuss in 3 weeks